### PR TITLE
Refactor WAL.Segments method to be part of the wal package

### PR DESCRIPTION
**Pull request by codesome**
It makes projects like Cortex consume the methods easily. I hope this small change looks fine.

It would be great if it can make it into `v2.15`. /cc @bwplotka

### CODE REVIEW

File: tsdb/wal/wal.go [hunk 1]
```diff
@@ -673,8 +673,8 @@ func (w *WAL) log(rec []byte, final bool) error {
 
 // Segments returns the range [first, n] of currently existing segments.
 // If no segments are found, first and n are -1.
-func (w *WAL) Segments() (first, last int, err error) {
```
Comment (roidelapluie) [inline tsdb/wal/wal.go]: I think that this function should now be reordered

File: tsdb/wal/wal.go [hunk 2]
```diff
@@ -260,7 +260,7 @@ func NewSize(logger log.Logger, reg prometheus.Registerer, dir string, segmentSi
 	}
 	w.metrics = newWALMetrics(w, reg)
 
-	_, last, err := w.Segments()
+	_, last, err := Segments(w.dir)
```
Comment (roidelapluie) [inline tsdb/wal/wal.go]: w.Dir()

**Comment by roidelapluie** [comment]
2 comments, otherwise LGTM

**Comment by brian-brazil** [comment]
@codesome Is this still relevant?

**Comment by codesome** [comment]
Yes. I will fix the reviews soon and lets get this merged.

File: tsdb/wal/wal.go (hunk 1 above)
Comment (codesome) [reply tsdb/wal/wal.go:676]: @roidelapluie What do you mean by reordering? Move it within the file?

Comment (roidelapluie) [reply tsdb/wal/wal.go:676]: Yes, it is in the middle of the func (w *WAL)  functions.

**Comment by codesome** [comment]
Merging this on green

**Pull request merged**